### ISSUE TITLE
testCompilation doesn't work with separate compilation anymore due to -Ytest-pickler

### ISSUE THREAD

**Issue by biboudis** [lampepfl/dotty#5103]
```
$ cp -r tests/pos/i4023 tests/pos/i4024
sbt> testCompilation i4023 
sbt> testCompilation i4024
 ```

First `testCompilation` is OK, second raises compilation/classpath error (i4023 is used for demonstration purposes). Reproduced on master on mine and @liufengyun's PC.

**Comment by biboudis** [comment]
It related to #5059 and the `compiler/test/dotc/pos-test-pickling.blacklist`. When the new directory is added there, testCompilation works again for this directory. cc @OlivierBlanvillain

**Comment by liufengyun** [comment]
Fixed in `#5571` -- in pickling tests, separate compilation only test the first group.
